api/proto/index/v2/index.proto

```diff
@@ -0,0 +1,161 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+package index.v2;
+option go_package = "./index/v2;index";
+
+// Service provides an external API for managing the index records
+service Service {
+  // Create allows to new records for a search.
+  rpc Create(CreateRecordsRequest) returns (CreateRecordsResult);
+  // Create allows to create a new records, streaming the records data in the request
+  rpc CreateWithStreamData(stream CreateIndexStreamRequest) returns (CreateRecordsResult);
+  // DeleteNode allows to delete a Node and all its children identified by the path provided
+  rpc DeleteNode(Path) returns (google.protobuf.Empty);
+  // ListNodes returns all known children for the Path provided
+  rpc ListNodes(Path) returns (Nodes);
+  // Patch allows to insert, update or delete an index's records
+  rpc PatchRecords(PatchRecordsRequest) returns (PatchRecordsResult);
+  // ListRecords returns list of records for a path associated with it
+  rpc ListRecords(ListRequest) returns (ListRecordsResult);
+  // Search runs the search across all indexes by the specified query. Result will
```

Comment:
Search runs the search across all the index records matching the query.

Reply:
+